core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -168,15 +178,17 @@ public TenantObject setTrustAnchor(final X509Certificate certificate) {
 
         try {
             final JsonObject trustedCa = new JsonObject()
+                    .put(TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, certificate.getSubjectX500Principal().getName())
                     .put(TenantConstants.FIELD_PAYLOAD_CERT, certificate.getEncoded());
-            return setProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA, trustedCa);
-        } catch (CertificateEncodingException e) {
+            return addTrustedCaConfiguration(trustedCa);
+        } catch (final CertificateEncodingException e) {
             throw new IllegalArgumentException("cannot encode certificate");
         }
     }
 
     /**
      * Gets the trusted certificate authority configured for this tenant.
+     * This method assumes that the tenant is configured with a single Trusted CA.
```

Comment:
actually, it only requires to have *at least one* CA configured

Reply:
since this method will eventually be replaced with `getTrustedCertificateAuthorities()`, i will remove this sentence